app/models/manageiq/providers/amazon/inventory/collector/target_collection.rb

```diff
@@ -131,72 +137,43 @@ def stack_template(stack_name)
 
   def parse_targets!
     target.targets.each do |t|
-      if t.kind_of?(::EmsEvent)
-        parse_parse_ems_event_target!(t)
-      elsif t.kind_of?(::Vm)
+      case t
+      when Vm
         parse_vm_target!(t)
       end
     end
   end
 
-  def parse_parse_ems_event_target!(t)
-    collect_references!(t.full_data.fetch_path("detail", "requestParameters") || {})
-    collect_references!(t.full_data.fetch_path("detail", "responseElements") || {})
-
-    instance_id = t.full_data.fetch_path("detail", "instance-id")
-    instances_refs << instance_id if instance_id
-  end
-
-  def collect_references!(hash)
-    instances_refs << hash["instanceId"] if hash["instanceId"]
-    private_images_refs << hash["imageId"] if hash["imageId"]
-    key_pairs_refs << hash["keyName"] if hash["keyName"]
-    stacks_refs << hash["stackId"] if hash["stackId"]
-    stacks_refs << hash["stackName"] if hash["stackName"]
-    cloud_networks_refs << hash["vpcId"] if hash["vpcId"]
-    cloud_subnets_refs << hash["subnetId"] if hash["subnetId"]
-    network_ports_refs << hash["networkInterfaceId"] if hash["networkInterfaceId"]
-    security_groups_refs << hash["groupId"] if hash["groupId"]
-    floating_ips_refs << hash["allocationId"] if hash["allocationId"]
-    load_balancers_refs << hash["loadBalancerName"] if hash["loadBalancerName"]
-    cloud_volumes_refs << hash["volumeId"] if hash["volumeId"]
-    cloud_volume_snapshots_refs << hash["snapshotId"] if hash["snapshotId"]
-
-    # TODO(lsmola) how to handle tagging? Tagging affects e.g. a name of any resource, but contains only a generic
-    # resourceID
-    # "requestParameters"=>
-    #   {"resourcesSet"=>{"items"=>[{"resourceId"=>"vol-07ad036724e3175a5"}]},
-    #    "tagSet"=>{"items"=>[{"key"=>"Name", "value"=>"ladas_volue_2"}]}},
-    # I think we can parse the resource id, so guess where it belongs.
-    # TODO(lsmola) RegisterImage, by creating image from a volume snapshot, should we track the block device
-    # mapping and refresh also snapshot?
-
-    collect_references!(hash["networkInterface"]) if hash["networkInterface"]
-
-    (hash.fetch_path("groupSet", "items") || []).each { |x| collect_references!(x) }
-    (hash.fetch_path("instancesSet", "items") || []).each { |x| collect_references!(x) }
-    (hash.fetch_path("instances") || []).each { |x| collect_references!(x) }
-    (hash.fetch_path("networkInterfaceSet", "items") || []).each { |x| collect_references!(x) }
-  end
-
   def parse_vm_target!(t)
-    instances_refs << t.ems_ref if t.ems_ref
+    target.add_target(:association => :vms, :manager_ref => {:ems_ref => t.ems_ref}) if t.ems_ref
```

Comment:
If  this is mostly always `target.add_target`, you might also allow `target.add` syntax